classes/UpgradeSelfCheck.php

```diff
@@ -409,13 +440,55 @@ private function checkMaxExecutionTime(): int
         return (int) @ini_get('max_execution_time');
     }
 
-    public function isPhpVersionCompatible(): bool
+    public function getPhpCompatibilityRange(): array
     {
-        if (!class_exists(ConfigurationTest::class)) {
-            return true;
+        if (null !== $this->phpCompatibilityRange) {
+            return $this->phpCompatibilityRange;
+        }
+
+        $targetVersion = $this->upgrader->version_num;
+
+        if (version_compare($targetVersion, '8', '<')) {
+            $targetVersion = VersionUtils::getPrestashopMinorVersion($targetVersion);
+
+            if (isset($this::PRESTASHOP_17_PHP_REQUIREMENTS[$targetVersion])) {
+                $range = $this::PRESTASHOP_17_PHP_REQUIREMENTS[$targetVersion];
+            } else {
+                return [];
+            }
+        } else {
+            try {
+                $range = $this->distributionApiService->getPhpVersionRequirements($targetVersion);
+            } catch (DistributionApiException $apiException) {
+                return [];
+            }
         }
+        $currentPhpVersion = VersionUtils::getHumanReadableVersionOf(PHP_VERSION_ID);
+        $range['php_current_version'] = $currentPhpVersion;
+
+        return $this->phpCompatibilityRange = $range;
+    }
+
+    /**
+     * @return int
+     */
+    public function getPhpRequirementsState()
+    {
+        $phpCompatibilityRange = $this->getPhpCompatibilityRange();
 
-        return (bool) ConfigurationTest::test_phpversion();
+        if (empty($phpCompatibilityRange)) {
+            return self::UNKNOWN_PHP_REQUIREMENTS;
+        } else {
+            $versionMin = VersionUtils::getPhpVersionId($phpCompatibilityRange['php_min_version']);
+            $versionMax = VersionUtils::getPhpVersionId($phpCompatibilityRange['php_max_version']);
+            $actualVersion = VersionUtils::getPhpMajorMinorVersionId();
```

Comment:
Ok for the wording, however I do not use `$phpCompatibilityRange['php_current_version']` voluntarily, because it contains the patch versions, which poses a problem in the comparison in certain cases

Reply:
For example if a store is in PHP 7.1.3, and the max is PHP 7.1, this will cause a problem when in reality it is good

Reply:
$phpCompatibilityRange['php_current_version'] is mainly present for the view

Reply:
Yes, I agree with you, forget about the other change. That is another comment I initially wanted to put and forgot to undo the diff when I decided it wasn't relevant.